src/dmd/target.d

```diff
@@ -314,6 +323,204 @@ extern (C++) struct Target
                 break;
         }
     }
+
+    /**
+     Sets CPU Operating System, and optionally C/C++ runtime environment from the given triple
+     e.g.
+        x86_64+avx2-apple-darwin20.3.0
+        x86-unknown-linux-musl-clang
+        x64-windows-msvc
+        x64-pc-windows-msvc
+     */
+    void setTriple(const(char)* _triple)
```

Comment:
Isn't it better to add all this new logic to a new struct, called `Triple` or something, and a function which parses the triple string and returns a new value of `Triple`. Then hopefully all these new functions can be made pure, `Triple` could be immutable/const and it would be easier to unit test.

Reply:
Done